### PR TITLE
Bitcode strip embedded iOS frameworks

### LINKED ISSUE #51904: hello_world_ios__compile bytesrelease_size_bytes size regression
https://github.com/flutter/flutter/pull/51453 appears to have increased the `hello_world_ios__compile bytesrelease_size_bytes` from 8537475 bytes to 164814091 bytes.

image

### PR DESCRIPTION
**Pull request by jmagman**
## Description

https://github.com/flutter/flutter/pull/51453 moved embedding logic from Xcode into the tool to give more flexibility.  When Xcode copies frameworks for build (not archive), it strips bitcode to reduce the size of the generated app.  It also doesn't copy Modules or Headers since the app only needs this for linking and not at runtime.  Copy both of these behaviors into xcode_backend (and someday the assemble dart scripts).

The archive size is no different before and after https://github.com/flutter/flutter/pull/51453, but reducing the release app binary is still nice to do for the sake of developer installation time.

## Related Issues

Fixes https://github.com/flutter/flutter/issues/51904.
I don't think this will completely get us back to the previous binary size, but it will be much closer.  However that test should be re-baselined for this change.  The real fix is to change that test to archive instead of release building, since release building and zippping is a very poor proxy for final release binary size.  See https://github.com/flutter/flutter/issues/40471.

## Tests

Renamed and expanded the purpose of ios_obfuscate_test.

I also tested by turning on bitcode on in the hello_world project and confirming the App Store Connect distribution options still had the "Include bitcode" checkbox enabled (this means all embedded content contained bitcode).

## Checklist
- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change
- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*

### CODE REVIEW

**Comment by jmagman** [comment]
\cc @jonahwilliams

File: packages/flutter_tools/bin/xcode_backend.sh [hunk 1]
```diff
@@ -292,14 +292,17 @@ EmbedFlutterFrameworks() {
...
 
   # Embed the actual Flutter.framework that the Flutter app expects to run against,
   # which could be a local build or an arch/type specific build.
-  # Remove it first since Xcode might be trying to hold some of these files - this way we're
-  # sure to get a clean copy.
-  RunCommand rm -rf -- "${xcode_frameworks_dir}/Flutter.framework"
-  RunCommand cp -Rv -- "${flutter_ios_engine_folder}/Flutter.framework" "${xcode_frameworks_dir}/"
+
+  # Copy Xcode behavior and don't copy over headers or modules.
+  RunCommand rsync -av --delete --filter "- .DS_Store/" --filter "- Headers/" --filter "- Modules/" "${flutter_ios_engine_folder}/Flutter.framework" "${xcode_frameworks_dir}/"
```
Comment (jmagman) [inline packages/flutter_tools/bin/xcode_backend.sh]: --filter instead of --exclude so missing patterns don't fail.

**Review by dnfield — COMMENTED** [review]
Makes sense - can we update an existing test to verify that we're doing this right?

**Comment by jmagman** [comment]
> Makes sense - can we update an existing test to verify that we're doing this right?

Renamed and expanded the purpose of ios_obfuscate_test.  Confirmed this test succeeded before https://github.com/flutter/flutter/pull/51453, fails on master, and passes on this PR.

**Review by dnfield — APPROVED** [review]
LGTM

**Pull request merged**